tests/test_request_overrider.js

```diff
@@ -13,731 +13,718 @@
+
+    const scope = nock('http://example.test')
+      .post('/', exampleRequestBody)
+      .reply()
+
+    const req = http.request({
       host: 'example.test',
-      method: 'GET',
+      method: 'POST',
       path: '/',
       port: 80,
-    },
-    res => {
+      headers: { Expect: '100-continue' },
+    })
+
+    const onData = sinon.spy()
+
+    req.on('response', res => {
+      expect(res.statusCode).to.equal(200)
+      // The `end` event will not fire without a `data` listener, though it
+      // will never fire since the body is empty. This is consistent with
+      // the Node docs:
+      // https://nodejs.org/api/http.html#http_class_http_clientrequest
+      res.on('data', onData)
       res.on('end', () => {
-        expect(res.complete).to.be.true()
+        expect(onData).not.to.have.been.called()
         scope.done()
-        t.end()
+        done()
       })
-      // Streams start in 'paused' mode and must be started.
-      // See https://nodejs.org/api/stream.html#stream_class_stream_readable
-      res.resume()
-    }
-  )
-  req.end()
-})
-
-test('Request with `Expect: 100-continue` triggers continue event', t => {
-  // This is a replacement for a wide-bracket regression test that was added
-  // for https://github.com/nock/nock/issues/256.
-  //
-  // The behavior was subsequently changed so 'continue' is emitted only when
-  // the `Expect: 100-continue` header is present.
-  //
-  // This test was adapted from this test from Node:
-  // https://github.com/nodejs/node/blob/1b2d3f7ae7f0391908b70b0333a5adef3c8cb79d/test/parallel/test-http-expect-continue.js#L35
-  //
-  // Related:
-  // https://tools.ietf.org/html/rfc2616#section-8.2.3
-  // https://github.com/nodejs/node/issues/10487
-  const exampleRequestBody = 'this is the full request body'
-
-  const scope = nock('http://example.test')
-    .post('/', exampleRequestBody)
-    .reply()
-
-  const req = http.request({
-    host: 'example.test',
-    method: 'POST',
-    path: '/',
-    port: 80,
-    headers: { Expect: '100-continue' },
-  })
-
-  const onData = sinon.spy()
-
-  req.on('response', res => {
-    expect(res.statusCode).to.equal(200)
-    // The `end` event will not fire without a `data` listener, though it
-    // will never fire since the body is empty. This is consistent with
-    // the Node docs:
-    // https://nodejs.org/api/http.html#http_class_http_clientrequest
-    res.on('data', onData)
-    res.on('end', () => {
-      expect(onData).not.to.have.been.called()
-      scope.done()
-      t.end()
     })
-  })
 
-  req.on('continue', () => {
-    req.end(exampleRequestBody)
+    req.on('continue', () => {
+      req.end(exampleRequestBody)
+    })
   })
-})
 
-// https://github.com/nock/nock/issues/1836
-test('when http.get and http.request have been overridden before nock overrides them, http.get calls through to the expected method', async t => {
-  // TODO Investigate why this is needed when it's also in the `afterEach()`
```

Comment:
This todo and `end` hook is what prompted the investigation into retorting Sinon and led to the update in setup.js.